examples/rmrk/resource/wasm/tests/resources.rs

```diff
@@ -165,6 +193,63 @@ fn adding_existing_part_to_resource_by_admin_succeeds() {
     }
 }
 
+#[tokio::test]
+async fn adding_existing_part_to_resource_by_admin_via_client_succeeds() {
+    // Arrange
+    let fixture = Fixture::new(ADMIN_ID);
+    let catalog_program_id = ActorId::from(fixture.catalog_program().id().into_bytes());
+    let resource_program_id = ActorId::from(fixture.resource_program_for_async().id().into_bytes());
+    let resource = resource_client::Resource::Composed(resource_client::ComposedResource {
+        src: "<src_uri>".into(),
+        thumb: "<thumb_uri>".into(),
+        metadata_uri: "<metadata_uri>".into(),
+        base: catalog_program_id,
+        parts: vec![1, 2, 3],
+    });
+    let _ = fixture
+        .add_resource_via_client(ADMIN_ID, RESOURCE_ID, resource, resource_program_id)
+        .await
+        .unwrap()
+        .unwrap();
+
+    let mut parts = BTreeMap::new();
+    parts.insert(
+        PART_ID,
+        Part::Fixed(FixedPart {
+            z: Some(1),
+            metadata_uri: "<metadata_uri>".into(),
+        }),
+    );
+    fixture.add_parts(ADMIN_ID, &parts);
+
+    // Act
+    let add_part_reply = fixture
+        .add_part_to_resource_via_client(ADMIN_ID, RESOURCE_ID, PART_ID, resource_program_id)
+        .await
+        .unwrap()
+        .unwrap();
+
+    // Assert
+    assert_eq!(PART_ID, add_part_reply);
```

Comment:
Instead of unwrapping at line 230
```suggestion
    assert_eq!(Ok(PART_ID), Ok(add_part_reply));
```